avm/res/logic/integration-account/certificate/main.bicep

```diff
@@ -0,0 +1,74 @@
+metadata name = 'Integration Account Certificates'
+metadata description = 'This module deploys an Integration Account Certificate.'
+
+@description('Required. The Name of the certificate resource.')
+param name string
+
+@description('Optional. Resource location.')
+param location string = resourceGroup().location
+
+@description('Conditional. The name of the parent integration account. Required if the template is used in a standalone deployment.')
+param integrationAccountName string
+
+@description('Optional. The key details in the key vault.')
+param key keyVaultKeyReferenceType?
```

Comment:
Why not use a resouce derived type `resourceInput<'Microsoft.Logic/integrationAccounts/certificates@2019-05-01'>.properties.key` ? :)